src/platform-implementation-js/namespaces/app-menu.ts

```diff
@@ -102,11 +104,85 @@ Items with CollapsiblePanels can also have accessories which provide primary act
 */
 export default class AppMenu {
   #driver;
+  #events = stream<
+    {
+      type: 'collapseToggled';
+      /** Whether or not the AppMenuBurger is open or not. */
+      open: boolean;
+      /**
+       * 'start' is when a `transactionstart` event is fired on a collapsible panel.
+       * 'end' is when a `transactioncancel` event is fired on a collapsible panel. */
+      stage: 'start' | 'end';
```

Comment:
could you give more context why the 'stage' is necessary?

couple of things:
1. the event named `collapseToggle` but the property is `open`, lets make it either `openChanged` event or `collapseToggled` with `collapsed` property
2. What is the purpose of `stage` prop? I would imagine the consumer code would do smth like 
```
const isOpened = (event.open == false && stage === 'start' || event.open == true && stage === 'end')
```
if that is the case, lets remove `stage` prop from this event, and do this calculation within AppMenu to avoid such logic being duplicated in all places where it is needed.
3. the `transitionend` event should still be listened too, otherwise if for some reason the transition will actually complete (like gmail will fix a bug or something), the `transitioncancel` will never be called.

Reply:
a few responses:

> the event named collapseToggle but the property is open, lets make it either openChanged event or collapseToggled with collapsed property

1. @dmy7r0 and I have gone back and forth on `collapsed` v. `open`. One thing that bit us during `AppMenu` development was stale values. https://github.com/InboxSDK/InboxSDK/pull/936#discussion_r1244346685 reminded me that the subscriber checking toggleCollapsed is probably a better idea.

> What is the purpose of stage prop?

2. At least in the use case we're trying to solve in the repo downstream, which you might've missed [the link for above this thread](https://github.com/StreakYC/MailFoo/pull/14017), we _only_ immediately care about `transitionend`. Riffing on [this comment](https://github.com/InboxSDK/InboxSDK/pull/936#discussion_r1244346685) more, just emitting the `TransitionEvent` and having the subscriber do what they want with `TransitionEvent.type` being `transition{start,end,cancel}` seems fairly straightforward.

> the transitionend event should still be listened too, otherwise if for some reason the transition will actually complete (like gmail will fix a bug or something), the transitioncancel will never be called.

3. Added in [079ff95](https://github.com/InboxSDK/InboxSDK/pull/936/commits/079ff957cebe68b578391319dd4e2ab8a3d74dc6) with the caveat that as of writing, it is not called. It looks like Gmail is removing the class with a transition time of 200ms something like 190ms into the animation right now.